src/firefly/js/util/VOAnalyzer.js

```diff
@@ -1152,6 +1152,11 @@ export function applyLinkSub(tableModel, href='', rowIdx, defval='') {
  * @returns {string}    the resolved href after subsitution
  */
 export function applyTokenSub(tableModel, val='', rowIdx, def) {
+
+    if (!val) {
+        return def;
+    }
+
     const vars = val.match && val.match(/\${[\w -.]+}/g);
```

Comment:
Good catch.  However, this will return `def` when val is a numeric 0.  Will you modify your change request to this:

> const vars = val?.match?.(/\${[\w -.]+}/g);

This should provide null-checking as well.  Do you agree?  Thanks.

Reply:
Aaah yes my bad, I've made a new commit with your suggestion. Also tested it out and it does indeed fix the exception I was seeing.